portfolio/src/main/webapp/script.js

```diff
@@ -12,12 +12,115 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+var context;
+var canvas;
+var myBubbleArray;
+var numBubs = 20;
+var animationID = null;
+// Resizes bubbles when page size is changed
+window.addEventListener('resize', init)
+
+function startChat() {
+  grabComments();
+  init();
+  grabUser();
+}
+
+class Bubble {
+  constructor(color) {
+    this.speed_factor = 3;
+    if (canvas.width < 900) {
+      this.speed_factor = 1.75;
+    }
+    this.color = color;
+    this.size = Math.random() * canvas.width;
+    this.x = Math.random() * canvas.width;
+    this.y = Math.random() * canvas.height;
+    this.radius = Math.floor(Math.random() * Math.sqrt(canvas.width));
+    this.dx = Math.random() * this.speed_factor * this.randomDirection();
+    this.dy = Math.random() * this.speed_factor * this.randomDirection();
+    this.opacity = Math.random();
+  }
+  drawBubble() {
+    context.beginPath();
+    context.fillStyle = this.color;
+    context.globalAlpha = this.opacity;
+    context.arc(this.x, this.y, this.radius, 0, Math.PI * 2, true);
+    context.closePath();
+    context.fill();
+  }
+  randomDirection() {
+    if (Math.random() > .5) {
+      return -1;
+    } else {
+      return 1;
+    }
+  }
+}
+
+function init() {
+  // cancels any previous animation to start a new one
+  if (animationID !== null) {
+    window.cancelAnimationFrame(animationID);
+    canvas.width = window.innerWidth;
+    canvas.height = window.innerHeight;
+  } else {
+    canvas = document.getElementById('myCanvas');
+    context = canvas.getContext('2d');
+    canvas.width = window.innerWidth;
+    canvas.height = window.innerHeight;
+  }
+  myBubbleArray = [];
```

Comment:
Use `const` or` let` to declare local variables (https://google.github.io/styleguide/jsguide.html#features-local-variable-declarations)

Reply:
Ping